### PR TITLE
Add CLI for Chill and refactor configuration to it them testable

### PR DESCRIPTION
**Pull request by kabirbaidhya**
Things done:
* CLI Chill 
* Refactor configuration
* Refactor email & twilio services
* Add more tests

### CODE REVIEW

File: src/monitoring/init.js [hunk 1]
```diff
@@ -0,0 +1,18 @@
+import merge from 'webpack-merge';
```
Comment (mesaugat) [inline src/monitoring/init.js]: ↔️

File: src/config/config.js [hunk 2]
```diff
@@ -1,14 +1,35 @@
...
-const loadedConfig = Yaml.load(CONFIG_FILE_NAME);
-const config = merge(defaultConfig, loadedConfig);
+export const CACHE_KEY = 'config';
+export const DEFAULT_FILENAME = 'chill.yml';
 
-export default config;
+/**
+ * Resolve configuration by reading the configuration file.
+ * 
+ * @param {string} [filename=DEFAULT_FILENAME]
```
Comment (mesaugat) [inline src/config/config.js]: `String`

File: test/services/email.test.js [hunk 3]
```diff
@@ -20,102 +20,79 @@ describe('email.isEnabled', () => {
...
+  });
+});
 
-    email.notify({
-      status: STATUS_UP,
-      name: randomName
-    });
+describe('email.notify', () => {
+  let sandbox;
+  // let email
```
Comment (mesaugat) [inline test/services/email.test.js]: ?

File: test/services/email.notify.back [hunk 4]
```diff
@@ -0,0 +1,130 @@
+import faker from 'faker';
+import sinon from 'sinon';
+import { assert } from 'chai';
+import proxyquire from 'proxyquire';
+// import logger from '../../src/utils/logger';
+import * as config from '../../src/config/config';
+// import * as email from '../../src/services/email';
+// import * as emailRenderer from '../../src/utils/emailRenderer';
+import { STATUS_UP } from '../../src/services/status';
+
```
Comment (mesaugat) [inline test/services/email.notify.back]: email.notify.back? Is this file not needed?

File: src/cli/index.js [hunk 5]
```diff
@@ -0,0 +1,21 @@
+let yargs = require('yargs');
```
Comment (mesaugat) [inline src/cli/index.js]: const

File: src/cli/index.js [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+let chill = require('../../package');
```
Comment (mesaugat) [inline src/cli/index.js]: const

File: src/cli/index.js [hunk 7]
```diff
@@ -0,0 +1,21 @@
...
+let name = 'chill';
```
Comment (mesaugat) [inline src/cli/index.js]: New line here?

File: src/services/email.js [hunk 8]
```diff
@@ -1,16 +1,29 @@
...
 import messages from '../common/messages';
-import emailClient from '../utils/emailClient';
-import { render } from '../utils/emailRenderer';
+import * as config from '../config/config';
+import * as emailRenderer from '../utils/emailRenderer';
+
+/**
+ * Get the nodemailer email client using the configured transport.
+ *
+ * @returns {Object} The nodemailer Transporter object.
```
Comment (mesaugat) [inline src/services/email.js]: No need of description here, fine though.

File: src/services/email.js [hunk 9]
```diff
@@ -48,22 +63,19 @@ export async function notify(params) {
...
+    text: messages[status].text(name, downtime)
+  });
+  let html = emailRenderer.render('status', message);
 
   return {
     from: sender,
     to: receivers,
-    html: emailBody,
-    subject: subject
+    html,
```
Comment (mesaugat) [inline src/services/email.js]: ↔️

File: src/services/hipchat.js [hunk 10]
```diff
@@ -42,26 +41,39 @@ export async function notify(params) {
...
-  const { status, name } = params;
-
+  let { status, name } = params;
   let { text, color } = messages[status];
 
   return {
     message: text(name, params.downtime),
     color: color,
-    from: 'Chill@noreply.com',
+    from: 'chill@noreply.com',
```
Comment (mesaugat) [inline src/services/hipchat.js]: ↔️ 😆

File: package.json [hunk 11]
```diff
@@ -1,6 +1,6 @@
 {
-  "name": "node-chill",
-  "version": "0.0.0",
+  "name": "chill-cli",
+  "version": "v1.0.0-alpha",
```
Comment (kabirbaidhya) [inline package.json]: 1.0.0-alpha

File: src/cli/index.js [hunk 12]
```diff
@@ -0,0 +1,27 @@
+import 'babel-polyfill';
+import yargs from 'yargs';
+import chill from '../../package';
+import init from '../monitoring/init';
+import * as config from '../config/config';
+
+let name = 'chill';
```
Comment (romitamgai) [inline src/cli/index.js]: const?

File: package.json [hunk 13]
```diff
@@ -43,16 +45,28 @@
...
     "chai": "^3.5.0",
+    "chai-subset": "^1.5.0",
     "codecov": "^2.1.0",
     "eslint": "^3.19.0",
     "faker": "^4.1.0",
     "istanbul": "1.1.0-alpha.1",
     "mocha": "^3.2.0",
     "nodemon": "^1.11.0",
     "npm-run-all": "^3.1.1",
+    "proxyquire": "^1.7.11",
```
Comment (mesaugat) [inline package.json]: Not used?

Comment (kabirbaidhya) [inline package.json]: It wasn't. But now it's used :wink:

**Pull request merged**